src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -120,7 +126,16 @@
             throw new IOException("Unsupported credential type: " + credentials.getClass().getName());
         }
 
-        return gb.build();
+        GitHub github = gb.build();
+        try {
+            github.checkApiUrlValidity();
```

Comment:
Well what was the actual stack trace encountered when you turned off your network and ran branch indexing?